01-17-crafting-system/crafting/CraftingStation.gd

```diff
@@ -0,0 +1,57 @@
+extends Node
+class_name CraftingStation
+"""
+Responsible to craft items based on recipes
+"""
+
+
+export var craftable_items = []
+
+
+func can_craft_item(itens : Array, recipe : Array) -> int:
+	"""
+	Returns the amount of a certain item that can be crafted given a recipe and an array of itens
+	"""
+	var amount : = 999999
+	var can_craft : = true
+	for ingredient in recipe:
+		var has_ingredient : = false
+		for item in itens:
+			if item.item == ingredient.item:
+				if item.amount >= ingredient.amount:
+					has_ingredient = true
+					var craftable_amount : = int(item.amount / ingredient.amount)
+					amount = craftable_amount if craftable_amount < amount else amount
+		can_craft = has_ingredient
+	if can_craft:
+		return amount
+	else:
+		return -1
+
+
+func get_craftables(itens : Array) -> Array:
+	"""
+	Returns an array of the possible craftable items with the given itens
+	"""
+	var craftables : = []
+	for craftable_item in craftable_items:
+		craftable_item = craftable_item as Item
```

Comment:
I wouldn't add that line for just one statement that uses it: `craftable_item.recipe`. At runtime, the items should already have the corresponding type so you're adding an extra operation.